tests/test_pipenv.py

```diff
@@ -440,6 +440,25 @@ def test_windows_pinned_pipfile(self):
             assert 'tablib' in p.lockfile['default']
 
 
+    @pytest.mark.e
+    @pytest.mark.install
+    @pytest.mark.vcs
+    def test_editable_vcs_install_in_pipfile_with_dependency_resolution_doesnt_traceback(self):
+        # See https://github.com/pypa/pipenv/issues/1240
+        with PipenvInstance() as p:
+            with open(p.pipfile_path, 'w') as f:
+                contents = """
+[packages]
+requests = {git = "https://github.com/requests/requests.git", editable = true}
+"oslo.utils" = "==1.4.0"
```

Comment:
So what actually is the dependency conflict here? Does `oslo.utils==1.4.0` conflict with requests? perhaps you can pin requests to a specific `ref` to ensure we keep having the same conflict? I'm fairly confident we will be changing resolver logic in the future so we may want to keep that in mind as it may invalidate this test (i.e. if we change the way resolver failures are handled/python version hacking operates).

Reply:
The dependency conflict is entirely contained within `oslo.utils`; it depends on a specific version of pbr, and the latest version of one of its dependencies depends on a different version of pbr.

So, really, it's just there to force a dependency resolution failure so that we end up on the broken code path, the second resolution attempt.

Reply:
Hrmm, so just because this state exists now doesn’t mean it will necessarily exist in the future. When it’s a package outside of our control, I’d prefer we avoid implicit testing conditions.

Is it possible for us to find package(s) we can pin at exact versions which will cause this failure mode?

Reply:
My latest commit switches to using requests, idna and the PyPA Sphinx theme (as it has no dependencies). It also adds a comment explaining why we're choosing these particular versions.